roles/openshift_health_checker/openshift_checks/package_availability.py

```diff
@@ -0,0 +1,69 @@
+from openshift_checks import OpenShiftCheck, OpenShiftCheckException
+from openshift_checks.mixins import NotContainerized
+
+
+class PackageAvailability(NotContainerized, OpenShiftCheck):
+    """Check that required RPM packages are available."""
+
+    name = "package_availability"
+    tags = ["preflight"]
+
+    def run(self, tmp, task_vars):
+        try:
+            rpm_prefix = task_vars["openshift"]["common"]["service_type"]
+        except (KeyError, TypeError):
+            raise OpenShiftCheckException("'openshift.common.service_type' is undefined")
```

Comment:
I agree with you, in this sense the checks are more like action plugins than modules.
The inherent difficulty is we need to have code somewhere that initializes the check with the proper information it needs, doing it on the loader/runner side makes things hard to deal with. Putting it at the control of the check developer allows for the uniform API.

If we create yet another layer for "check instantiation" (that would feed in to a check information like `openshift.common.service_type` without leaking the whole `task_vars`), then check developers would need to code both the check and the instantiation (probably taking data from `task_vars`, since that's the only input to the action plugin).